src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+        private attachmentService: AttachmentService,
+        private attachmentUnitService: AttachmentUnitService,
+        private alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.route.data.subscribe((data) => {
+            if ('attachment' in data) {
+                this.attachment = data.attachment;
+                if (this.attachment?.id) {
+                    this.attachmentService.getAttachmentFile(this.attachment.id).subscribe({
+                        next: (blob: Blob) => this.loadPdf(URL.createObjectURL(blob)),
+                        error: (error) => onError(this.alertService, error),
+                    });
+                } else {
+                    this.alertService.error('artemisApp.attachment.pdfPreview.attachmentIDError');
+                }
+            } else if ('attachmentUnit' in data) {
+                this.attachmentUnit = data.attachmentUnit;
+                if (this.attachmentUnit?.id) {
+                    this.attachmentUnitService.getAttachmentFile(this.attachmentUnit.id).subscribe({
+                        next: (blob: Blob) => this.loadPdf(URL.createObjectURL(blob)),
+                        error: (error) => onError(this.alertService, error),
+                    });
+                } else {
+                    this.alertService.error('artemisApp.attachment.pdfPreview.attachmentUnitIDError');
+                }
+            }
+        });
+        document.addEventListener('keydown', this.handleKeyboardEvents);
+        window.addEventListener('resize', this.resizeCanvasBasedOnContainer);
+    }
+
+    ngOnDestroy() {
+        document.removeEventListener('keydown', this.handleKeyboardEvents);
+        window.removeEventListener('resize', this.resizeCanvasBasedOnContainer);
+    }
+
+    private async loadPdf(fileUrl: string) {
+        try {
+            const loadingTask = PDFJS.getDocument(fileUrl);
+            const pdf = await loadingTask.promise;
+            this.totalPages = pdf.numPages;
+
+            for (let i = 1; i <= pdf.numPages; i++) {
+                const page = await pdf.getPage(i);
+                const viewport = page.getViewport({ scale: 1 });
+                const canvas = this.createCanvas(viewport);
+                const context = canvas.getContext('2d');
+                if (context) {
+                    await page.render({ canvasContext: context, viewport }).promise;
+                }
+
+                const container = this.createContainer(canvas, i);
+                this.pdfContainer.nativeElement.appendChild(container);
+            }
+
+            URL.revokeObjectURL(fileUrl);
+        } catch (error) {
+            onError(this.alertService, error);
+        }
+    }
+
+    private createCanvas(viewport: PDFJS.PageViewport): HTMLCanvasElement {
+        const canvas = document.createElement('canvas');
+        canvas.height = viewport.height;
+        canvas.width = viewport.width;
+        const fixedWidth = 250;
```

Comment:
What's the meaning of 250 here?

Reply:
I wanted to set the small slide page size to 250px. How can I define this kind of size better, on the top maybe or a different format?

Reply:
I think it would be fine if you defined a constant (`static readonly DEFAULT_SLIDE_WIDTH` or similar) for this